Algorithms.Tests/Crypto/Digests/Md2DigestTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Text;
+using Algorithms.Crypto.Digests;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Crypto.Digesters;
+
+public class Md2DigestTests
+{
+    private readonly Md2Digest digest = new Md2Digest();
+
+    private readonly string[] messages =
+    {
+        "",
+        "a",
+        "abc",
+        "message digest",
+        "abcdefghijklmnopqrstuvwxyz",
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789",
+        "12345678901234567890123456789012345678901234567890123456789012345678901234567890",
+    };
+
+    private readonly string[] expectedDigests =
+    {
+        "8350E5A3E24C153DF2275C9F80692773",
+        "32EC01EC4A6DAC72C0AB96FB34C0B5D1",
+        "DA853B0D3F88D99B30283A69E6DED6BB",
+        "AB4F496BFB2A530B219FF33031FE06B0",
+        "4E8DDFF3650292AB5A4108C3AA47940B",
+        "DA33DEF2A42DF13975352846C30338CD",
+        "D5976F79D83D3A0DC9806C3C66F3EFD8",
+    };
+
+    [Test]
+    public void Digest_ReturnsCorrectValue()
```

Comment:
Please use `TestCase` attribute to supply inputs

Reply:
I really think that there is an issue with how codecov checks the code at this point.

First there is the issue where lines 202-204 were not covered.

The condition on the `while` loop `(mBufferOffset != 0) && (length > 0)` was not met by the tests I've written.

This condition checks if there is any data left in the buffer (`mBufferOffset != 0`) and if there is any data left to process (`length = 0`). If both conditions are true, it processes the remaining data in the buffer before moving on to the next chunk of data.

To cover these lines, I created a test where `mBufferOffset` is not zero and `length` is greater than zero at the same time. This can be achieved by providing an input that is not a multiple of 16 (the size of the buffer), because the `Update` method processes data in chunks of 16 bytes.

For that reason, I added the following test case

```csharp
[TestCase("123456789012345678901234567890123456789012345678901234567890123456789012345678901", "ExpectedHash")]
```

The string has a length of 81, which is not a multiple of 16. Therefore, after processing 5 full chunks of 16 bytes, there will be a remainder of 1 byte in the buffer, which should trigger the `while((mBufferedOffset != 0) && (length > 0)) loop in the `Update` method.